tests/testthat/test-table_with_settings.R

```diff
@@ -0,0 +1,128 @@
+table_r <- reactive({
+  l <- rtables::basic_table() %>%
+    rtables::split_cols_by("Species") %>%
+    rtables::analyze(c("Sepal.Length"))
+  rtables::build_table(l, iris)
+})
+
+testthat::test_that("table_with_settings_srv: assertions", {
+  args <- list(
+    list(table_r = "er"),
+    list(show_hide_signal = TRUE, table_r = table_r)
+  )
+  for (arg in args) {
+    testthat::expect_error(
+      shiny::testServer(
+        teal.widgets:::table_with_settings_srv,
+        args = arg
+      ),
+      "Assertion"
+    )
+  }
+})
+
+testthat::test_that("table_with_settings_srv: hide works", {
```

Comment:
We are testing that it does not render any error but not if it works, correct?

Reply:
Yes, and I don't know how to test it differently with `testServer`

Reply:
Ok whether the plot is shown or not is to be tested by `shinytest2`. The intention here is that setting `show_hide_signal` to FALSE does not break the functionality and still returns a table html which is hidden.